packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -83,6 +74,14 @@ export class AwsCliCompatible {
         tokenCodeFn,
       });
     }
+
+    function iniFileCredentialFactories(theProfile: string) {
+      return [
+        () => profileCredentials(theProfile),
```

Comment:
Hello, is there a reason why credentials from the profile would take precedence here over the SSO? It's different than e.g. behavior of JS SDK v3 - https://github.com/aws/aws-sdk-js-v3/blob/31e72ac6063fdb3393c2d5042b8964238ce1b23a/packages/credential-provider-node/src/defaultProvider.ts#L52-L54

Or am I missing something here?